test/test-lex.cpp

```diff
@@ -221,27 +345,25 @@ TEST(test_lex, lex_invalid_big_int_number) {
                 VariantWith<error_big_int_literal_contains_decimal_point>(_)));
       });
 
-  // Complain about both the decimal point and the leading 0 digit.
+  /* // Complain about both the decimal point and the leading 0 digit. */
   check_tokens_with_errors(
-      u8"01.2n", {token_type::number},
+      u8"01.2n", {token_type::number,token_type::number},
```

Comment:
Hmm, I don't like this change, but if it's difficult to preserve this behavior, I'll tolerate it.